lib/evmone/instructions.hpp

```diff
@@ -1068,6 +1084,33 @@ inline TermResult return_impl(StackTop stack, int64_t gas_left, ExecutionState&
 inline constexpr auto return_ = return_impl<EVMC_SUCCESS>;
 inline constexpr auto revert = return_impl<EVMC_REVERT>;
 
+inline TermResult returncontract(
+    StackTop stack, int64_t gas_left, ExecutionState& state, code_iterator pos) noexcept
+{
+    const auto& offset = stack[0];
+    const auto& size = stack[1];
+
+    if (state.msg->kind != EVMC_CREATE3)
```

Comment:
In case we end up allowing EOF creation transactions with empty `to` (as opposed to relying only on transaction to Creator Contract), this will have to be adjusted to allow `RETURNCONTRACT` in the context of creation transaction.